### PR TITLE
#169817815 Users should be able to log in using Facebook or Google

### PR DESCRIPTION
**Pull request by jo-rdan**
#### What does this PR do?
It implements the functionality of signing in with social accounts
#### Description of Task to be completed?
- add a button that calls the google consent screen
- add a Facebook button
#### How should this be manually tested?
- clone this repo on your local computer
- make sure you have all the packages installed by typing `npm install` in the terminal
- start the server by running `npm start` then browse to this route` localhost:8080/auth` and click on google and it should send the consent screen, choose an account and it will redirect you on the homepage
#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[Starts #169817815](https://www.pivotaltracker.com/story/show/169817815)
#### Screenshots (if appropriate)
ezgif com-optimize

**When Redirected**
image

#### Questions:

### CODE REVIEW

File: src/components/auth/SocialLogin.jsx [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React, { Component } from 'react';
```
Comment (Ruqoyah) [inline src/components/auth/SocialLogin.jsx]: I didn't see the test for this component.

**Comment by Ruqoyah** [comment]
Screenshot 2020-02-06 at 12 46 54 PM
The button should look like what's on the mockup. Kindly update

**Comment by Ruqoyah** [comment]
Screenshot 2020-02-06 at 12 51 33 PM
Also, I tried logging in with facebook but I got the above error in the screenshot

**Comment by Ruqoyah** [comment]
Also, kindly note this implementation is meant to be available on signup and login according to the Mock up that has already been presented to the PO.

**Comment by jo-rdan** [comment]
> Screenshot 2020-02-06 at 12 46 54 PM
> 
> The button should look like what's on the mockup. Kindly update

@Ruqoyah the design of the buttons are being implemented by Benit who is working on the sign-in feature, 
I worked on the functionality so that if the sign-in feature is complete they will just call the links

**Review by mujohn26 — APPROVED** [review]

File: src/components/Welcome.jsx [hunk 2]
```diff
@@ -1,16 +1,28 @@
 import React, { Component } from 'react';
 import { Link } from "react-router-dom";
+import queryString from 'query-string';
 
 class Welcome extends Component {
+
+    componentDidMount() {
+        const query = queryString.parse(location.search);
+        const data = JSON.parse(query.info || '{}');
+        localStorage.setItem('token', data.token);
```
Comment (Jajabenit250) [inline src/components/Welcome.jsx]: @jo-rdan  good progress so far, but I would like if you don't mind instead of saving user token as token on local storage why not considering saving it as user because am saving user token as user or if you feel naming them as token is a better idea we can stick to it too. let me know what you think.

File: src/components/Welcome.jsx [hunk 3]
```diff
@@ -1,16 +1,28 @@
...
         return (
-        <div>
-            <h1>You are welcome at Barefoot nomad</h1>
-			<Link to='/login'><h1>Click me to login</h1></Link>
-            <Link to='/redux'><h1>Click check redux</h1></Link>
-		</div>
+            <div>
+                <h1>You are welcome at Barefoot nomad</h1>
+                <Link to='/login'><h1>Click me to login</h1></Link>
+                <Link to='/auth'><h1>Click me to login with social accounts</h1></Link>
```
Comment (Jajabenit250) [inline src/components/Welcome.jsx]: am wondering why you change /auth to be a main route of application. my suggestion is that we can stick on /   empty route and redirect user who is not already signed in to login page and those who signed in and already have a valid token to the dashboard.

**Review by Jajabenit250 — COMMENTED** [review]
Good Progress so far, Kindly check the feedback left.

File: src/components/Welcome.jsx (hunk 2 above)
Comment (jo-rdan) [reply src/components/Welcome.jsx]: @Jajabenit250 thank you for your feedback, however, I think naming it token can be best because we need to specify what it is. what do you think?

File: src/components/Welcome.jsx (hunk 3 above)
Comment (jo-rdan) [reply src/components/Welcome.jsx]: @Jajabenit250 No, it is not the home link, that is the link for navigating to the social sign-in buttons. we still redirect the user to the homepage if he is already signed in.

File: src/components/Welcome.jsx (hunk 2 above)
Comment (Jajabenit250) [reply src/components/Welcome.jsx]: @jo-rdan okay let user token I think it is better.

File: src/components/Welcome.jsx (hunk 3 above)
Comment (Jajabenit250) [reply src/components/Welcome.jsx]: @jo-rdan  Jordan my point is if user is already signed in and we have token in local storage that are valid can he use your route again? I was thinking maybe remove everything from local storage at the beginning of your component because I can see in your implementation you are setting token to local storage but in case I access the route again I can still set token while there is a token which already exists in local storage. or check if token in local storage is valid and then redirect him to dashboard if not valid give him access to using signing in using social auth.

Comment (jo-rdan) [reply src/components/Welcome.jsx]: @Jajabenit250 I understand your concern but you are working on the sign-in feature, I think that you are in charge of that because if you really think about it my feature is just another part of your feature, my buttons will appear in your overall component, I worry about the functionality of the social authentication

Comment (Jajabenit250) [reply src/components/Welcome.jsx]: @jo-rdan okay I get that. that's why I was suggesting if posting change that /auth route to / route. because /auth is no use to your feature. if you are only working on functionality.

**Review by Jajabenit250 — APPROVED** [review]

File: src/components/Welcome.jsx (hunk 2 above)
Comment (Ruqoyah) [inline src/components/Welcome.jsx]: This token is never getting set. Kindly look into that.

File: jest.config.js [hunk 4]
```diff
@@ -97,7 +97,7 @@ module.exports = {
   // A map from regular expressions to module names that allow to stub out resources with a single module
   moduleNameMapper: {
     '\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
-      '<rootDir>/__mocks__/fileMock.js',
+      'jest-transform-stub',
```
Comment (Ruqoyah) [inline jest.config.js:101]: Why did you change this?

File: src/components/Welcome.jsx [hunk 5]
```diff
@@ -1,16 +1,28 @@
...
+
     render() {
+
         return (
-        <div>
-            <h1>You are welcome at Barefoot nomad</h1>
-			<Link to='/login'><h1>Click me to login</h1></Link>
-            <Link to='/redux'><h1>Click check redux</h1></Link>
-		</div>
+            <div>
```
Comment (Ruqoyah) [inline src/components/Welcome.jsx]: What's going on in this lines of code below? Are we using them for anything?

File: src/components/App.jsx [hunk 6]
```diff
@@ -4,6 +4,7 @@ import DefaultLayout from '../layouts/default.layout.jsx';
 import MainLayout from '../layouts/main.layout.jsx';
 
 class App extends Component {
+	
 	render() {
 		return (
 			<div className='app'>
```
Comment (Ruqoyah) [inline src/components/App.jsx:10]: Just like what @Jajabenit250 said, there is no need to have /auth route, once I visit localhost://8080, it should just redirect me to the login page where you have the social icon buttons and @Jajabenit250 would add the remaining elements to the page.

File: src/components/auth/SocialLogin.jsx [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
```
Comment (Ruqoyah) [inline src/components/auth/SocialLogin.jsx]: Since you are using a functional component, there is no need to call Component here, you can just import it this way `import React from 'react';`

File: src/components/App.jsx (hunk 6 above)
Comment (jo-rdan) [reply src/components/App.jsx:10]: @Ruqoyah, you mean the login should be like a landing page?

Comment (Ruqoyah) [reply src/components/App.jsx:10]: I checked through the mockup on Figma and found out the landing page is the login page (I hope I am correct) so I am saying if I visit localhost://8080, it should redirect me to localhost://8080/login

Comment (jo-rdan) [reply src/components/App.jsx:10]: No, we don't actually have a dashboard, we started working on signup and login. login is not the dashboard

File: src/index.js [hunk 8]
```diff
@@ -1,13 +1,9 @@
...
-import { Provider } from 'react-redux';
-import store from './store/index';
 import App from './components/App.jsx';
 import './styles/index.scss';
 
 ReactDOM.render(
-	<Provider store={store}>
-		<App />
-	</Provider>,
+		<App />,
```
Comment (Ruqoyah) [inline src/index.js]: Why did you take out the store connection?

File: src/helpers/verifyToken.js [hunk 9]
```diff
@@ -0,0 +1,12 @@
+import { decode } from 'jsonwebtoken';
+import dotenv from 'dotenv';
+
+dotenv.config();
+
+const verifyToken = (token) => {
+   const decodedToken = decode(token);
+   console.log(decodedToken);
```
Comment (Ruqoyah) [inline src/helpers/verifyToken.js]: Take out the console.log()

File: src/components/auth/SocialLogin.jsx [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+      textTransform: 'capitalize'
+   }
+});
+
+const setToken = () => {
+   localStorage.setItem('token', 'token');
+}
+const SocialLogin = () => {
+   const classes = useStyles();
+   // useEffect(() => {
```
Comment (Ruqoyah) [inline src/components/auth/SocialLogin.jsx]: Take out the commented code.

File: jest.config.js (hunk 4 above)
Comment (Ruqoyah) [reply jest.config.js:101]: I am asking a question on this but you didn't respond.

**Review by Ruqoyah — COMMENTED** [review]
Good Job Jordan! However, you didn't fully do what I asked you to do. When I visit localhost:8080/, let it redirect me to localhost:8080/auth/signin, what you did only redirect me to the signin page when I visit localhost:8080/auth, what happens if user visit localhost:8080 or https://barefoot-namad.com, all they would see is a blank page.

Note: Benit is also working on the signin page, and I have given him feedback on what I noticed on his as well, I believe you guys should work together to get this fix so you can do the same thing. Whoever implement all the feedback first, I will merge his PR then the other person can get the update from develop.

**Comment by jo-rdan** [comment]
> Good Job Jordan! However, you didn't fully do what I asked you to do. When I visit localhost:8080/, let it redirect me to localhost:8080/auth/signin, what you did only redirect me to the signin page when I visit localhost:8080/auth, what happens if user visit localhost:8080 or https://barefoot-namad.com, all they would see is a blank page.
> 
> Note: Benit is also working on the signin page, and I have given him feedback on what I noticed on his as well, I believe you guys should work together to get this fix so you can do the same thing. Whoever implement all the feedback first, I will merge his PR then the other person can get the update from develop.

Yes I was fixing that now!

File: src/index.js (hunk 8 above)
Comment (jo-rdan) [reply src/index.js]: I was not using the redux store, actions or reducers. So, when it is merged others have everything it will have the store back.

**Comment by Ruqoyah** [comment]
Awesome @jo-rdan looking forward to it.

**Comment by jo-rdan** [comment]
> Good Job Jordan! However, you didn't fully do what I asked you to do. When I visit localhost:8080/, let it redirect me to localhost:8080/auth/signin, what you did only redirect me to the signin page when I visit localhost:8080/auth, what happens if user visit localhost:8080 or https://barefoot-namad.com, all they would see is a blank page.
> 
> Note: Benit is also working on the signin page, and I have given him feedback on what I noticed on his as well, I believe you guys should work together to get this fix so you can do the same thing. Whoever implement all the feedback first, I will merge his PR then the other person can get the update from develop.

the reason I changed the `jest.config.js` is that whoever set up this file, `<rootDir>/__mocks__/fileMock.js` this path lead to nothing therefore adding an image wrecked the tests

File: src/index.js (hunk 8 above)
Comment (Ruqoyah) [reply src/index.js]: @jo-rdan you don't need to delete them even if you aren't using it. it doesn't affect your work, just leave the files there, just the way they are.

Comment (Ruqoyah) [reply src/index.js]: Don't forget someone worked on setting up redux, so you removing it simply means, you are destroying what the person work on.

Comment (jo-rdan) [reply src/index.js]: Ok I will put them back
